DDPB-4363: Update dev set up docs (#852)
ministryofjustice/opg-digideps: README.md

# Complete the deputy report

This app is the [Complete the deputy report][service] service. It provides an online reporting service used by deputies

> to submit their reports, and the private area for case managers to review submitted reports.

Reviewer: I think substituting private for admin here makes sense as that's the actual name of the service. Or maybe:

```suggestion
This app is the [Complete the deputy report][service] service. It provides an online reporting service that has a publicly available frontend used by deputies
to submit their reports, and a VPN restricted admin area for case managers to review submitted reports.
```

## Requirements

You must have Docker installed. Once installed click on the cog icon in Docker, select `Experimental Features` and tick

> both boxes. To complete the setup, locate the Digideps root directory in your terminal and run `make up-app-build` or

Reviewer 2: Re Docker. Worth noting these boxes are only available on more recent MacOS versions.

> You must have Docker installed. Once installed click on the cog icon in Docker, select `Experimental Features` and tick
> both boxes. To complete the setup, locate the Digideps root directory in your terminal and run `make up-app-build` or
> the equivalent docker command `docker-compose up -d --build --remove-orphans` which can found in the `Makefile`.

Reviewer: Suggestions:

```suggestion
You must have Docker installed. If your MacBook contains Apple Silicon (e.g. a minimum of an M1 processor) you can enable a much faster implementation of the filesystem in Docker for Mac. In Docker, click on the cog icon in Docker, select `Experimental Features` and tick both boxes.

To complete the setup, locate the Digideps root directory in your terminal and run `make up-app-build` or the equivalent docker command `docker-compose up -d --build --remove-orphans` which can found in the `Makefile`.
```

> If developing the app then ensure you have [pre-commit](https://pre-commit.com/) installed to take advantage of the
> pre-commit [hooks](.pre-commit-config.yaml) we've added to the project to make PRs a more consistent and enjoyable
> experience.

Reviewer: Similar to above - when the markdown is rendered it will wrap as required so hard newlines aren't needed.

```suggestion
If developing the app then ensure you have [pre-commit](https://pre-commit.com/) installed to take advantage of the pre-commit [hooks](.pre-commit-config.yaml) we've added to the project to make PRs a more consistent and enjoyable experience.
```

## Installation

- Add `127.0.0.1 digideps.local admin.digideps.local api.digideps.local www.digideps.local` to `/etc/hosts`
- Navigate to the root directory of this repository and run `make up-app`

> - Check `https://digideps.local/` (Deputy area) and `https://admin.digideps.local/` (Admin area). Your browser will warn
>   you about a self-signed certificate.

Reviewer: ```suggestion
- Check `https://digideps.local/` (Deputy area) and `https://admin.digideps.local/` (Admin area). Your browser will warn you about a self-signed certificate.
```

![Digideps traffic flow diagram](./docs/traffic_flow_diagram.png)

> This diagram can be updated in `traffic_flow_diagram.puml` and then rendered to PNG
> using `plantuml ./docs/traffic_flow_diagram.puml -o .`.

Reviewer: ```suggestion
This diagram can be updated in `traffic_flow_diagram.puml` and then rendered to PNG using `plantuml ./docs/traffic_flow_diagram.puml -o .`.
```

## Runbook

> Our runbook, incident response process and other OPG technical guidance can be
> found [here](https://ministryofjustice.github.io/opg-technical-guidance/#opg-technical-guidance).

Reviewer: ```suggestion
Our runbook, incident response process and other OPG technical guidance can be found [here](https://ministryofjustice.github.io/opg-technical-guidance/#opg-technical-guidance).
```
